### PR TITLE
[xDAI] Make EthPricing native token agnostic

### PR DESCRIPTION
**Pull request by fleupold**
Part of 1418

EthPricing is really trying to estimate the price of the native token (ETH in the case of mainnet, xDAI in the case of DAI). This token is not always at ID 0.

This PR makes the token ID configurable and renames it to native_token to be compatible with non eth based side chains.

### Test Plan
Unit tests.

### CODE REVIEW

File: driver/src/main.rs [hunk 1]
```diff
@@ -239,6 +239,11 @@ struct Options {
     /// the startup time.
     #[structopt(long, env = "ORDERBOOK_FILE", parse(from_os_str))]
     orderbook_file: Option<PathBuf>,
+
+    /// Use an orderbook file for persisting an event cache in order to speed up
+    /// the startup time.
```
Comment (e00E) [inline driver/src/main.rs:244]: Accidentally copied doc string.

File: price-estimator/src/main.rs [hunk 2]
```diff
@@ -133,6 +133,11 @@ struct Options {
         case_insensitive = true,
     )]
     economic_viability_strategy: EconomicViabilityStrategy,
+
+    /// Use an orderbook file for persisting an event cache in order to speed up
+    /// the startup time.
```
Comment (e00E) [inline price-estimator/src/main.rs:138]: Here too

**Review by e00E — APPROVED** [review]

File: contracts/artifacts/BatchExchange.json [hunk 3]
```diff
@@ -1 +1 @@
...
...d in previous epoch - token was received from exchange in current auction batch"}}},"title":"BatchExchange - A decentralized exchange for any ERC20 token as a multi-token batch auction with uniform clearing prices. For more information visit: <https://github.com/gnosis/dex-contracts>"},"networks":{"1":{"address":"0x6F400810b62df8E13fded51bE75fF5393eaa841F","links":{"IdToAddressBiMap":"0xED4d05496C71e71cC2A8726af1242C22108d1761","IterableAppendOnlySet":"0xCDDB32b6Bb2808D5B5115dAab207479cE98d2636"},"transactionHash":"0xe74e8e965afc67f96a38caee794f008be33a7fe7ea96b9baad7b2963250ac36a"},"100":{"address":"0x66b89680A34394Cd73A4964AF60f56c9669C2452","links":{"IdToAddressBiMap":"0x048E53A455a058462eA58442E1D94Fbc955495cB","IterableAppendOnlySet":"0x57E6B987c2ccd421859A244dd22a0D5a62D88f91"},"transactionHash":"0x61c641256f7ca31f95f2e8a96e4a641758ceff3213905baf35efe5cc5a73256f"},"4":{"address":"0xC576eA7bd102F7E476368a5E98FA455d1Ea34dE2","links":{"IdToAddressBiMap":"0x5c4C6bf91240A5fdBfB9a1BEd8d43227046e2feA","IterableAppendOnlySet":"0x0D47D0548FDAD66B06E81a826EED8c687aCddBCB"},"transactionHash":"0xc5bf4f48747093a79b623c2a2392ccbdb73a3d788e7d1f0f53f640ea18496d90"}},"userdoc":{"methods":{}}}
```
Comment (nlordell) [inline contracts/artifacts/BatchExchange.json:1]: Was this updated with the contract `vendor` script? https://github.com/gnosis/dex-services/tree/master/contracts#vendor-script

**Review by nlordell — APPROVED** [review]
Looks good. Just a question about how the vendored ABIs were updated.

File: contracts/artifacts/BatchExchange.json (hunk 3 above)
Comment (fleupold) [reply contracts/artifacts/BatchExchange.json:1]: Yes , I will revert this. I was pulling in a new dex-contract version that had the updated addresses.

**Review by bh2smith — COMMENTED** [review]
I'm a little late to the party, but this is an awesome change! Great idea.

**Pull request merged**